metals/src/main/scala/scala/meta/internal/metals/BloopServers.scala

```diff
@@ -121,6 +125,207 @@ final class BloopServers(
+    messageActionItem match {
+
+      case item
+          if item == Messages.BloopGlobalJsonFilePremodified.applyAndRestart =>
+        writeJVMPropertiesToBloopGlobalJsonFile(
+          bloopGlobalJsonFilePath,
+          bloopCreatedByMetalsFilePath,
+          requestedBloopJvmProperties,
+          maybeJavaHome
+        ) match {
+          case Failure(exception) => Future.failed(exception)
+          case Success(_) =>
+            shutdownServer()
+            reconnect().ignoreValue
+        }
+
+      case item
+          if item == Messages.BloopGlobalJsonFilePremodified.openGlobalJsonFile =>
+        val position = new Position(0, 0)
+        val range = new org.eclipse.lsp4j.Range(position, position)
+        val command = ClientCommands.GotoLocation.toExecuteCommandParams(
+          ClientCommands.WindowLocation(
+            bloopGlobalJsonFilePath.toURI.toString,
+            range
+          )
+        )
+        Future.successful(languageClient.metalsExecuteClientCommand(command))
+
+      case item
+          if item == Messages.BloopGlobalJsonFilePremodified.useGlobalFile =>
+        Future.successful(())
+    }
+  }
+
+  private def updateBloopGlobalJsonFileThenRestart(
+      bloopGlobalJsonFilePath: AbsolutePath,
+      bloopCreatedByMetalsFilePath: AbsolutePath,
+      requestedBloopJvmProperties: List[String],
+      maybeJavaHome: Option[String],
+      reconnect: () => Future[BuildChange]
+  ): Future[Unit] = {
+    writeJVMPropertiesToBloopGlobalJsonFile(
+      bloopGlobalJsonFilePath,
+      bloopCreatedByMetalsFilePath,
+      requestedBloopJvmProperties,
+      maybeJavaHome
+    ) match {
+      case Failure(exception) => Future.failed(exception)
+      case Success(_) =>
+        languageClient
+          .showMessageRequest(
+            Messages.BloopJvmPropertiesChange.params()
+          )
+          .asScala
+          .flatMap {
+            case messageActionItem
+                if messageActionItem == Messages.BloopJvmPropertiesChange.reconnect =>
+              shutdownServer()
+              reconnect().ignoreValue
+            case _ =>
+              Future.successful(())
+          }
+    }
+
+  }
+
+  private def getBloopGlobalJsonFilePath(): Option[AbsolutePath] =
+    Try {
+      AbsolutePath(
+        Paths
+          .get(System.getProperty("user.home"))
+          .resolve(".bloop/bloop.json")
+      )
+    }.toOption
+
+  private def getBloopCreatedByMetalsFilePath(): Option[AbsolutePath] =
+    Try {
+      AbsolutePath(
+        Paths
+          .get(System.getProperty("user.home"))
+          .resolve(".bloop/created_by_metals.lock")
+      )
+    }.toOption
+
+  private def getBloopGlobalJsonLastModifiedTime(
+      bloopGlobalJsonFilePath: AbsolutePath
+  ): Long =
+    Try {
+      bloopGlobalJsonFilePath.toNIO.toFile.lastModified()
+    }.getOrElse(0)
+
+  /**
+   * First we check if the user requested to update the Bloop JVM properties through the extension.
```

Comment:
Then why didn’t scalafmt do it automatically this time? I am sure I ran it…
Can you please provide a tad bit of more hints?

Reply:
Sure, so with no settings Scalafmt will wrap your comments for you, but part of that is dropping them down so:

```scala
/** my docs
 * @param foo
 *      it drops the explanation of foo down here instead of leaving. it above
 */
```
So because it drops it down and (I personally) find that super annoying since it wastes a line, we decided to put the following setting in our `.scalafmt.conf` file to avoid it:

```
docstrings.wrap = no
```
But there is a trade off, because now even when our comments are super long, it won't wrap them.

Reply:
So `docstrings.wrap = yes` ?

Reply:
No no, don't change the config. Just manually wrap your lines if possible so they aren't super super wide. Like

```
* here is my super super long thing that is super long long and if I now go wider than my screen might allow so I might want to wrap
```
become
```
* here is my super super long thing that is super long long and if I
* now go wider than my screen might allow so I might want to wrap
```

Reply:
LOL

Reply:
done